### ISSUE TITLE
Bug: ATM function does not work. “Payment failed, use a different wallet”

### ISSUE THREAD

**Issue by AxelHamburch** [lnbits/lnurldevice#49]
The ATM function worked perfectly with LNURLDevice version 0.6.5 and LNbits version 0.12.7. After updating my LNbits server to 0.12.10 I noticed that the ATM function no longer works. The Lightning Wallet (no matter which one) shows the message: “Payment failed, use a different wallet”.

I am familiar with this somewhat general message when my LNbits server switches to void wallet, but this is not the case here. Deposits and withdrawals from the wallet in which the LNURL device extension is installed work, only the request for withdrawal by the ATM reports this error. 

To rule out a server, I tested the ATM function of the demo.lnbits.com server. It has version 0.12.9 and shows the same error. It seems to be an error that has nothing to do with the LNURLDevice extension directly, but must have crept in via the LNbits version.

The payments are then displayed as pending in LNbits and new attempts are rejected as already claimed. However, the payments are never made. 

The error must have occurred after LNbits version 0.12.7. As of 0.12.9 it is definitely there, as a test with the demo server shows.

**Comment by AxelHamburch** [comment]
Addendum, unfortunately I didn't notice anything suspicious in the logs.

**Comment by AxelHamburch** [comment]
I think I have found the error. After trying several LNbits versions 0.12.8 to 0.12.10 and also testing different LNPoS / ATM devices, I accidentally came across the “withdraw” button of the hanging transaction. 

![01 024-08-21 20_16_28-ATM Testing Wallet - LNbits (DEMO SERVER)](https://github.com/user-attachments/assets/d2ecd2b6-a58e-497d-95ab-c824af20f5f3)

What I got was this error message.

![02 2024-08-21 20_16_54-LNbits (DEMO SERVER)](https://github.com/user-attachments/assets/c4b54e16-033f-4560-8abe-e2a234348700)

So I installed the “Withdraw Links” extension. And lo and behold, suddenly the ATM function also worked. I uninstalled it again and got the error message “Payment failed, use a different wallet.”

![photo_2024-08-21_20-44-46](https://github.com/user-attachments/assets/9ab2f52c-a87d-4b99-a363-0c45787df505)

I don't know why, but as of version 0.12.8 something must have been changed that connects and requires the LNURLDevice ATM function with the Withdraw Links extension, otherwise the payment will not take place.

**Comment by AxelHamburch** [comment]
I was able to reproduce the error with LNbits version 0.12.12 and LNURLdevice 0.6.5 (the highest available for LNbits 0.12.12). The ATM function did not work and always shows the message “Payment failed, use a different wallet.”. Only when I manually added the LNURLdevice version 0.6.8 and installed it did it work again. There seems to have been a temporary incompatibility between the LNbits / LNURLdevice versions. The only annoying thing is that 0.12.12 was delivered with an LNURLdevice version that did not work. I will close the issue now.
